testingii/src/App.js

```diff
@@ -0,0 +1,104 @@
+import React, { Component } from 'react';
+import './App.css';
+import Display from './display/Display.js'
+import Dashboard from './dashboard/Dashboard.js'
+
+class App extends Component {
+  state = {
+    balls: 0,
+    strikes: 0,
+    outs: 0 
+  }
+
+  onStrike = (e) => {
```

Comment:
Your onStrike function has a few logical and syntactical errors popping up. First of all, you need to prevent the default behavior, which you forgot to do. Then, you have to remember that each time you press the button, it renders the function from the top-down.  So the order of your logic matters. Therefore, if you prefer to nest all of your functionality (which is syntactically perfectly legit), you should restructure this function to look more like this in order to achieve the desired functionality for MVP:

```suggestion
  onStrike = (e) => {
    e.preventDefault(); //prevent default behavior
    let { outs, strikes, balls } = this.state;

    if (outs === 2 && strikes == 2) {
      alert("Yooour out! New Inning!");
      this.setState({
        strikes: 0,
        balls: 0,
        outs: 0
      });
    } else if (outs < 3 && strikes < 2) {
      //handle strikes
      this.setState({
        strikes: strikes + 1
      });
    }
    //handle strike out
    else if (outs < 3 && strikes === 2) {
      alert("Strike Out!");
      this.setState({ outs: outs + 1, strikes: 0, balls: 0 });
    }
  };
```

This solves all of the errors and then the correct alerts pop up at the correct times. Make sense?